benchmark/cublas/blas2/syr2.cpp

```diff
@@ -0,0 +1,188 @@
+/**************************************************************************
+ *
+ *  @license
+ *  Copyright (C) Codeplay Software Limited
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  For your convenience, a copy of the License has been included in this
+ *  repository.
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ *
+ *  SYCL-BLAS: BLAS implementation using SYCL
+ *
+ *  @filename syr2.cpp
+ *
+ **************************************************************************/
+
+#include "../utils.hpp"
+
+template <typename scalar_t>
+std::string get_name(std::string t, int n, scalar_t alpha) {
+  std::ostringstream str{};
+  str << "BM_Syr2<" << blas_benchmark::utils::get_type_name<scalar_t>() << ">/"
+      << t << "/" << n << "/" << alpha;
+  return str.str();
+}
+
+template <typename scalar_t, typename... args_t>
+static inline void cublas_routine(args_t&&... args) {
+  if constexpr (std::is_same_v<scalar_t, float>) {
+    CUBLAS_CHECK(cublasSsyr2(std::forward<args_t>(args)...));
+  } else if constexpr (std::is_same_v<scalar_t, double>) {
+    CUBLAS_CHECK(cublasDsyr2(std::forward<args_t>(args)...));
+  }
+  return;
+}
+
+template <typename scalar_t>
+void run(benchmark::State& state, cublasHandle_t* cuda_handle_ptr,
+         std::string uplo, index_t n, scalar_t alpha, bool* success) {
+  // Standard test setup.
+  const char* uplo_str = uplo.c_str();
+
+  index_t lda = n;
+  index_t incX = 1;
+  index_t incY = 1;
+
+  // The counters are double. We convert m and n to double to avoid
+  // integer overflows for n_fl_ops and bytes_processed
+  scalar_t size_d = static_cast<scalar_t>(n * (n + 1) / 2);
+
+  state.counters["n"] = static_cast<double>(size_d);
```

Comment:
this should be `static_cast<double>(n)` right?

Reply:
I am following the same convention of `spr` in syclblas, where it uses `size_d` as counter. We could decide to use `n` in all symmetric operators or use the actual `size_d` and set it as standard, ok?